media/js/views/modals.js

```diff
@@ -176,4 +201,67 @@
         }
     });
 
+    window.LCB.GiphyModalView = Backbone.View.extend({
+        events: {
+            'keypress .search-giphy': 'stopReturn',
+            'keyup .search-giphy': 'loadGifs'
+        },
+        initialize: function(options) {
+            this.render();
+        },
+        render: function() {
+            this.$el.on('shown.bs.modal hidden.bs.modal',
+                        _.bind(this.refresh, this));
+        },
+        refresh: function() {
+            this.$el.find('.giphy-results ul').empty();
+            this.$('.search-giphy').val('').focus();
+        },
+        stopReturn: function(e) {
+            if(e.keyCode === 13) {
+                return false;
+            }
+        },
+        loadGifs: _.throttle(function() {
```

Comment:
Throttle generates lots of unnecessary traffic, as the requests for the old images are not aborted and the GIFs tend to be rather big. Consider using debounce to hit the API when the user has stopped typing.